### PR TITLE
don't allocate memory so often in ClusterComm requests

### PR DESCRIPTION
**Pull request by jsteemann**
when creating requests in the ClusterComm. previously, we have
created several strings and other objects in ClusterComm::a?syncRequest
and then moved or copied them into Communicator::addRequest.

Communicator::addRequest has then created a NewRequest object from that,
while holding a mutex.

this NewRequest instance was later picked up by Communicator::work_once
and was move into Communicator::createRequestInProgress. In here, an
object of type RequestInProgress was created from it on heap. And this
object was finally passed to an object of type CurlHandle.
So we definitely moved and copied things around a lot...

Now, we create an object of type NewRequest in the ClusterComm, while
not holding any locks. This object is wrapped into a unique_ptr, which
is later on only moved around into its final place. Any internal data
from NewRequest is never moved or copied now. When the RequestInProgress
instance is created, it simply takes over ownership for the NewRequest
instance.

Additionally a few unused copies were removed elsewhere, and the
creation of a 1024 byte empty StringBuffer per request has been
eliminated.

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3688/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3689/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3712/

File: lib/SimpleHttpClient/Communicator.cpp [hunk 1]
```diff
@@ -455,132 +455,127 @@ void Communicator::handleResult(CURL* handle, CURLcode rc) {
...
+        }
+        break;
+      case CURLE_WRITE_ERROR:
+        if (rip->_aborted) {
           callErrorFn(rip, TRI_COMMUNICATOR_REQUEST_ABORTED, {nullptr});
-          break;
-        default:
-          LOG_TOPIC(ERR, arangodb::Logger::FIXME) << "curl return " << rc;
+        } else {
+          LOG_TOPIC(ERR, arangodb::Logger::FIXME)
```
Comment (danielhlarkin) [inline lib/SimpleHttpClient/Communicator.cpp]: Should we change the log topic on these to cluster comm?

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3729/

**Comment by jsteemann** [comment]
@danielhlarkin : log topics changed in Communicator.cpp

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3738/

**Pull request merged**